cli/tests/test_concurrent_operations.rs

```diff
@@ -227,6 +227,115 @@ fn test_concurrent_snapshot_wc_reloadable() {
+    for j in 0..200 {
+        let filename = format!("file_{j}");
+        let content = format!("content for file {j}\n").repeat(100);
+        work_dir.write_file(&filename, &content);
+    }
+    work_dir.run_jj(["commit", "-m", "initial"]).success();
+
+    // Create additional commits to iterate through with jj next/prev
+    for i in 0..10 {
+        for j in 0..50 {
+            let filename = format!("file_{i}_{j}");
+            let content = format!("content for commit {i} file {j}\n").repeat(100);
+            work_dir.write_file(&filename, &content);
+        }
+        work_dir
+            .run_jj(["commit", "-m", &format!("commit {i}")])
+            .success();
+    }
+
+    // Extract environment from TestEnvironment to use in spawned processes
+    let base_cmd = test_env.new_jj_cmd();
+    let jj_bin = base_cmd.get_program().to_owned();
+    let base_env: Vec<_> = base_cmd
+        .get_envs()
+        .map(|(k, v)| (k.to_owned(), v.map(|v| v.to_owned())))
+        .collect();
+
+    let repo_path = work_dir.root().to_path_buf();
+    let duration = std::time::Duration::from_secs(5);
+    let start = std::time::Instant::now();
+
+    let snapshot_handle = {
+        let jj_bin = jj_bin.clone();
+        let repo_path = repo_path.clone();
+        let base_env = base_env.clone();
+        std::thread::spawn(move || {
+            while start.elapsed() < duration {
+                // Snapshot repeatedly without delay
+                let mut cmd = std::process::Command::new(&jj_bin);
+                cmd.current_dir(&repo_path);
+                cmd.args(["debug", "snapshot"]);
+                for (key, value) in &base_env {
+                    if let Some(value) = value {
+                        cmd.env(key, value);
+                    }
+                }
+                let _ = cmd.output();
+            }
+        })
+    };
+
+    let next_handle = {
+        let jj_bin = jj_bin.clone();
+        let repo_path = repo_path.clone();
+        let base_env = base_env.clone();
+        std::thread::spawn(move || {
+            let mut count = 0;
+            while start.elapsed() < duration {
+                // Go through commit history back and forth
+                let direction = if count % 20 < 10 { "prev" } else { "next" };
+                let mut cmd = std::process::Command::new(&jj_bin);
+                cmd.current_dir(&repo_path);
+                cmd.arg(direction);
+                for (key, value) in &base_env {
+                    if let Some(value) = value {
+                        cmd.env(key, value);
+                    }
+                }
+                let _ = cmd.output();
+                count += 1;
+            }
+        })
+    };
+
+    snapshot_handle.join().unwrap();
+    next_handle.join().unwrap();
+
+    // Check for divergent operations
+    let output = work_dir.run_jj(["op", "log", "-T", "description"]);
+    let divergent_count = output
```

Comment:
nit: maybe `concurrent_count` is better? `divergent` made me think of divergent commits. While divergent commits are often created by concurrent operations, I think what we care about here is concurrent operations.